src/dune_rules/scope.ml

```diff
@@ -273,8 +273,21 @@ module DB = struct
           Some (project, stanzas))
       |> Path_source_map_traversals.parallel_map
            ~f:(fun _dir (project, stanzas) ->
+             let* host =
+               match host_context with
+               | None -> Memo.return None
+               | Some host_context ->
+                 let+ scope =
+                   let+ scopes, _public_libs_host =
+                     create_from_stanzas host_context
+                   in
+                   find_by_project scopes project
+                 in
+                 Some scope.db
+             in
              let+ db =
-               create_db_from_stanzas stanzas ~parent:public_libs ~lib_config
+               create_db_from_stanzas stanzas ~parent:public_libs ~host
```

Comment:
Thanks for restructuring. I think we can also take advantage of laziness here and make `host` lazily evaluated. So you would wrap `host` here with `Memo.lazy` and change the type in `Lib.DB.t` to `host : Memo.Lazy.t t option`.

Reply:
yup, makes sense. Made it a `Memo.Lazy.t`